src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionWithNestedFlavors.cs

```diff
@@ -70,34 +73,46 @@ public sealed override Task<IEnumerable<SuggestedActionSet>> GetActionSetsAsync(
             {
                 var extensionManager = this.Workspace.Services.GetService<IExtensionManager>();
 
-                _nestedFlavors = extensionManager.PerformFunction(
-                    Provider, CreateAllFlavors,
-                    defaultValue: ImmutableArray<SuggestedActionSet>.Empty);
+                _nestedFlavors = await extensionManager.PerformFunctionAsync(
+                    Provider, () => CreateAllFlavorsAsync(cancellationToken),
+                    defaultValue: ImmutableArray<SuggestedActionSet>.Empty).ConfigureAwait(false);
             }
 
             Contract.ThrowIfTrue(_nestedFlavors.IsDefault);
-            return Task.FromResult<IEnumerable<SuggestedActionSet>>(_nestedFlavors);
+            return _nestedFlavors;
         }
 
-        private ImmutableArray<SuggestedActionSet> CreateAllFlavors()
+        private async Task<ImmutableArray<SuggestedActionSet>> CreateAllFlavorsAsync(CancellationToken cancellationToken)
         {
             var builder = ArrayBuilder<SuggestedActionSet>.GetInstance();
 
-            var previewChangesSuggestedActionSet = GetPreviewChangesFlavor();
-            builder.Add(previewChangesSuggestedActionSet);
+            var primarySuggestedActionSet = await GetPrimarySuggestedActionSetAsync(cancellationToken).ConfigureAwait(false);
+            builder.Add(primarySuggestedActionSet);
 
-            if (_additionalFlavors != null)
+            if (_fixAllFlavors != null)
             {
-                builder.Add(_additionalFlavors);
+                builder.Add(_fixAllFlavors);
             }
 
             return builder.ToImmutableAndFree();
         }
 
-        private SuggestedActionSet GetPreviewChangesFlavor()
+        private async Task<SuggestedActionSet> GetPrimarySuggestedActionSetAsync(CancellationToken cancellationToken)
         {
+            using var _ = ArrayBuilder<SuggestedAction>.GetInstance(out var suggestedActions);
             var previewChangesAction = PreviewChangesSuggestedAction.Create(this);
-            return new SuggestedActionSet(categoryName: null, actions: ImmutableArray.Create(previewChangesAction));
+            suggestedActions.Add(previewChangesAction);
+
+            var refineUsingCopilotAction = await RefineUsingCopilotSuggestedAction.TryCreateAsync(this, cancellationToken).ConfigureAwait(false);
```

Comment:
what are the performance implications of this.  Are we now making lightbulbs slower because we're async waiting here on something potentially expensive.  that is def concerning.